base/interactiveutil.jl

```diff
@@ -61,9 +61,10 @@ function edit(path::AbstractString, line::Integer=0)
     end
 
     if is_windows() && name == "open"
-        systemerror(:edit, ccall((:ShellExecuteW,"shell32"), stdcall, Int,
-                                 (Ptr{Void}, Cwstring, Cwstring, Ptr{Void}, Ptr{Void}, Cint),
-                                 C_NULL, "open", path, C_NULL, C_NULL, 10) ≤ 32)
+        @static is_windows() && # don't emit this ccall on other platforms
```

Comment:
Right. I couldn't see an equivalent way to write this without doing more contortions, so this seemed the most readable way. I then added a comment there, so hopefully that'll be enough hint that the repeated conditional is intentional so nobody tries to delete it next week :P